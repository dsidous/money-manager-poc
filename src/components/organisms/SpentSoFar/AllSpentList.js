import React from 'react';
import { View, StyleSheet } from 'react-native';
import { StyledText } from '../../atoms';

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 20,
  },
  label: {
    fontSize: 23,
  },
  value: {
    fontSize: 24,
  },
});

const AllSpentList = ({ data }) => data.map(item => (
  <View key={item.title} style={styles.container}>
    <StyledText style={styles.label}>{item.title}</StyledText>
    <StyledText style={styles.value}>{`Â£${item.amount}`}</StyledText>
  </View>
));

export default AllSpentList;
