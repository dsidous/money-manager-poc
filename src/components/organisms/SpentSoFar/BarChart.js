import React, { PureComponent } from 'react';
import { View, StyleSheet } from 'react-native';
import { Bar } from 'react-native-pathjs-charts';
import { connect } from 'react-redux';
import { FETCH_BILL_MONITOR } from '../../../store/constants';

const styles = StyleSheet.create({
  container: {
    height: 300,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

const options = {
  width: 300,
  height: 200,
  margin: {
    top: 20,
    left: 25,
    bottom: 50,
    right: 20,
  },
  color: '#2980B9',
  gutter: 20,
  animate: {
    type: 'oneByOne',
    duration: 200,
    fillTransition: 3,
  },
  axisX: {
    showAxis: true,
    showLines: true,
    showLabels: true,
    showTicks: true,
    zeroAxis: false,
    orient: 'bottom',
    label: {
      fontFamily: 'Arial',
      fontSize: 8,
      fontWeight: true,
      fill: '#34495E',
      rotate: 45,
    },
  },
  axisY: {
    showAxis: true,
    showLines: true,
    showLabels: true,
    showTicks: true,
    zeroAxis: false,
    orient: 'left',
    label: {
      fontFamily: 'Arial',
      fontSize: 8,
      fontWeight: true,
      fill: '#34495E',
    },
  },
};

class BarChart extends PureComponent {
  componentDidMount() {
    this.props.dispatch({ type: FETCH_BILL_MONITOR });
  }

  render() {
    if (!this.props.billData) {
      return null;
    }
    return (
      <View style={styles.container}>
        <Bar data={this.props.billData} options={options} accessorKey="value" />
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    billData: state.spending.billData,
  };
};

export default connect(mapStateToProps)(BarChart);
