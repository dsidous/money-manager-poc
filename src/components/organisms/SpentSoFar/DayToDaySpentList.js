import React from 'react';
import { View, StyleSheet } from 'react-native';
import { StyledText } from '../../atoms';

const styles = StyleSheet.create({
  container: {
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 20,
  },
  title: {
    fontSize: 23,
  },
  amount: {
    fontSize: 24,
  }
});

const DayToDaySpentList = ({ data, sliceColor }) => {
  return data.map((item, i) => (
    <View key={i} style={styles.container}>
      <View style={{ flexDirection: 'row', alignItems: 'center' }}>
        <StyledText style={{ color: sliceColor[i], marginRight: 10 }}>&#9679;</StyledText>
        <StyledText style={styles.title}>{item.category}</StyledText>
      </View>
      <StyledText style={styles.amount}>{`Â£${item.amount}`}</StyledText>
    </View>
  ));
};

export default DayToDaySpentList;
