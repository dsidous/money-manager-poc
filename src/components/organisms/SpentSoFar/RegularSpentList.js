import React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
import { StyledText } from '../../atoms';

const styles = StyleSheet.create({
  container: {
    alignItems: 'flex-end',
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 20,
  },
  containerLeft: {
    paddingVertical: 5,
  },
  date: {
    fontSize: 18,
  },
  title: {
    fontSize: 23,
  },
  amount: {
    fontSize: 24,
  }
});

const RegularSpentList = ({ data, sliceColor, showHistory }) => data.map((item, i) => (
  <TouchableOpacity
    key={i}
    onPress={() => showHistory(item)}
  >
    <View style={styles.container}>
      <View style={{ flexDirection: 'row' }}>
        <View style={{ alignSelf: 'center' }}>
          <Text style={{ color: sliceColor[i], marginRight: 10 }}>&#9679;</Text>
        </View>

        <View style={styles.containerLeft}>
          <StyledText style={styles.date}>{item.date}</StyledText>
          <StyledText style={styles.title}>{item.title}</StyledText>
        </View>
      </View>
      <View>
        <StyledText style={styles.amount}>{`Â£${item.amount}`}</StyledText>
      </View>
    </View>
  </TouchableOpacity>
));

export default RegularSpentList;
