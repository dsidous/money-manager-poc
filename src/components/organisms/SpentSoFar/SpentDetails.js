import React, { Component } from 'react';
import { ScrollView, View, Text, Modal } from 'react-native';
import { connect } from 'react-redux';
import PieChart from 'react-native-pie-chart';
import AllSpentList from './AllSpentList';
import RegularSpentList from './RegularSpentList';
import DayToDaySpentList from './DayToDaySpentList';
import SpentHistory from './SpentHistory';


class SpentDetails extends Component {

  constructor() {
    super();

    this.state = {
      showModal: false,
    };
  }

  showHistory = (item) => {
    this.setState({ showModal: true });
  }

  hideHistory = () => {
    this.setState({ showModal: false });
  }

  render() {
    const data = this.props.spent[this.props.selected];
    const sliceColor = ['#F44336', '#2196F3', '#FFEB3B', '#4CAF50', '#FF9800'];
    const amounts = this.props.selected === 'allSpending'
      ? data.filter(item => item.title !== 'All').map(item => item.amount)
      : data.map(item => item.amount);
    let spendingList;

    if (this.props.selected === 'allSpending') {
      spendingList = <AllSpentList data={data} sliceColor={sliceColor} />;
    } else if (this.props.selected === 'regularSpending') {
      spendingList = <RegularSpentList data={data} sliceColor={sliceColor} showHistory={this.showHistory} />;
    } else {
      spendingList = <DayToDaySpentList data={data} sliceColor={sliceColor} />;
    }

    return (
      <View style={{ flex: 1 }}>
        <View style={{
          flex: 0.5,
          justifyContent: 'flex-start',
          alignContent: 'center',
          alignSelf: 'stretch',
          width: '100%',
        }}>
          <PieChart
            chart_wh={150}
            series={amounts}
            sliceColor={sliceColor}
            doughnut
            coverRadius={0.65}
            coverFill="#eee"
            style={{ alignSelf: 'center' }}
          />
        </View>
        <ScrollView
          contentContainerStyle={{ flexGrow: 1 }}
          style={{ flex: 4 }}
        >
          {spendingList}
        </ScrollView>

        <Modal
          animationType="slide"
          transparent={false}
          visible={this.state.showModal}
          onRequestClose={() => {
            Alert.alert('Modal has been closed.');
          }}>
          <SpentHistory hideHistory={this.hideHistory} />
        </Modal>
      </View>
    )
  };
};

const mapsStateToProps = (state) => {
  return {
    spent: state.spending.data,
  };
};

export default connect(mapsStateToProps)(SpentDetails);
