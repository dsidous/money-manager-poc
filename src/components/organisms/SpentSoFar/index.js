import React, { PureComponent } from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { StyledView } from '../../atoms';
import SpentDetails from './SpentDetails';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 25,
    justifyContent: 'flex-start',
  },
  headerContainer: {
    paddingBottom: 20,
    marginVertical: 20,
    borderBottomColor: '#ccc',
    borderBottomWidth: 1,
  },
  headerText: {
    fontSize: 18,
    fontWeight: 'bold',
  },
  buttonsView: {
    flexDirection: 'row',
    alignContent: 'center',
    alignSelf: 'center',
    marginVertical: 20,
  },
  buttons: {
    backgroundColor: '#ccc',
    borderWidth: 1,
    borderBottomWidth: 4,
    height: 60,
    justifyContent: 'center',
    paddingHorizontal: 30,
  },
  buttonsLabel: {
    fontSize: 20,
    color: '#000',
  },
  selectedButton: {
    backgroundColor: '#529FBE',
  },
  selectedButtonLabel: {
    color: '#fff'
  }
});
export default class SpentSoFar extends PureComponent {

  state = {
    selectedView: 'allSpending'
  }

  renderButtons = () => {
    const buttons = [
      {
        label: 'All',
        value: 'allSpending'
      },
      {
        label: 'Regular',
        value: 'regularSpending'
      },
      {
        label: 'Day to day',
        value: 'dayToDaySpending'
      },
    ]

    return buttons.map(item => {
      const selected = this.state.selectedView === item.value ? styles.selectedButton : '';
      const selectedLabel = this.state.selectedView === item.value ? styles.selectedButtonLabel : '';

      return (
        <TouchableOpacity
          key={item.label}
          style={[styles.buttons, selected]}
          onPress={() => this.setState({ selectedView: item.value })}
        >
          <Text style={[styles.buttonsLabel, selectedLabel]}>{item.label}</Text>
        </TouchableOpacity>
      )
    })

  }

  render() {

    return (
      <View style={styles.container}>
        <View style={styles.headerContainer}>
          <Text style={styles.headerText}>Spent so far</Text>
        </View>
        <StyledView style={styles.buttonsView}>
          {this.renderButtons()}
        </StyledView>
        <View style={{ flex: 6 }}>
          <SpentDetails selected={this.state.selectedView} />
        </View>
      </View>
    );
  }
}
