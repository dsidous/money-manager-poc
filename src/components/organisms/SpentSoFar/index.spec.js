import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import SpentSoFar from './index';

configure({ adapter: new Adapter() });

describe('spent so far', () => {
  const render = shallow(<SpentSoFar />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });
});
