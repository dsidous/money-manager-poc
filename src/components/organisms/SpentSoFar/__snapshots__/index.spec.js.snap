// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`spent so far renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SpentSoFar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "borderBottomColor": "#ccc",
              "borderBottomWidth": 1,
              "marginVertical": 20,
              "paddingBottom": 20,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Spent so far
          </Text>
        </Component>,
        <Styled(Component)
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "marginVertical": 20,
            }
          }
        >
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#ccc",
                  "borderBottomWidth": 4,
                  "borderWidth": 1,
                  "height": 60,
                  "justifyContent": "center",
                  "paddingHorizontal": 30,
                },
                Object {
                  "backgroundColor": "#529FBE",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#fff",
                  },
                ]
              }
            >
              All
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#ccc",
                  "borderBottomWidth": 4,
                  "borderWidth": 1,
                  "height": 60,
                  "justifyContent": "center",
                  "paddingHorizontal": 30,
                },
                "",
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 20,
                  },
                  "",
                ]
              }
            >
              Regular
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.2}
            onPress={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "#ccc",
                  "borderBottomWidth": 4,
                  "borderWidth": 1,
                  "height": 60,
                  "justifyContent": "center",
                  "paddingHorizontal": 30,
                },
                "",
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#000",
                    "fontSize": 20,
                  },
                  "",
                ]
              }
            >
              Day to day
            </Text>
          </TouchableOpacity>
        </Styled(Component)>,
        <Component
          style={
            Object {
              "flex": 6,
            }
          }
        >
          <Connect(SpentDetails)
            selected="allSpending"
          />
        </Component>,
      ],
      "style": Object {
        "flex": 1,
        "justifyContent": "flex-start",
        "paddingHorizontal": 25,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontSize": 18,
                "fontWeight": "bold",
              }
            }
          >
            Spent so far
          </Text>,
          "style": Object {
            "borderBottomColor": "#ccc",
            "borderBottomWidth": 1,
            "marginVertical": 20,
            "paddingBottom": 20,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "Spent so far",
            "ellipsizeMode": "tail",
            "style": Object {
              "fontSize": 18,
              "fontWeight": "bold",
            },
          },
          "ref": null,
          "rendered": "Spent so far",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#ccc",
                    "borderBottomWidth": 4,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 30,
                  },
                  Object {
                    "backgroundColor": "#529FBE",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ]
                }
              >
                All
              </Text>
            </TouchableOpacity>,
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#ccc",
                    "borderBottomWidth": 4,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 30,
                  },
                  "",
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    "",
                  ]
                }
              >
                Regular
              </Text>
            </TouchableOpacity>,
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#ccc",
                    "borderBottomWidth": 4,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 30,
                  },
                  "",
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    "",
                  ]
                }
              >
                Day to day
              </Text>
            </TouchableOpacity>,
          ],
          "style": Object {
            "alignContent": "center",
            "alignSelf": "center",
            "flexDirection": "row",
            "marginVertical": 20,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "All",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ]
                }
              >
                All
              </Text>,
              "onPress": [Function],
              "style": Array [
                Object {
                  "backgroundColor": "#ccc",
                  "borderBottomWidth": 4,
                  "borderWidth": 1,
                  "height": 60,
                  "justifyContent": "center",
                  "paddingHorizontal": 30,
                },
                Object {
                  "backgroundColor": "#529FBE",
                },
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "All",
                "ellipsizeMode": "tail",
                "style": Array [
                  Object {
                    "color": "#000",
                    "fontSize": 20,
                  },
                  Object {
                    "color": "#fff",
                  },
                ],
              },
              "ref": null,
              "rendered": "All",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Regular",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    "",
                  ]
                }
              >
                Regular
              </Text>,
              "onPress": [Function],
              "style": Array [
                Object {
                  "backgroundColor": "#ccc",
                  "borderBottomWidth": 4,
                  "borderWidth": 1,
                  "height": 60,
                  "justifyContent": "center",
                  "paddingHorizontal": 30,
                },
                "",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Regular",
                "ellipsizeMode": "tail",
                "style": Array [
                  Object {
                    "color": "#000",
                    "fontSize": 20,
                  },
                  "",
                ],
              },
              "ref": null,
              "rendered": "Regular",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Day to day",
            "nodeType": "class",
            "props": Object {
              "activeOpacity": 0.2,
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    "",
                  ]
                }
              >
                Day to day
              </Text>,
              "onPress": [Function],
              "style": Array [
                Object {
                  "backgroundColor": "#ccc",
                  "borderBottomWidth": 4,
                  "borderWidth": 1,
                  "height": 60,
                  "justifyContent": "center",
                  "paddingHorizontal": 30,
                },
                "",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Day to day",
                "ellipsizeMode": "tail",
                "style": Array [
                  Object {
                    "color": "#000",
                    "fontSize": 20,
                  },
                  "",
                ],
              },
              "ref": null,
              "rendered": "Day to day",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(SpentDetails)
            selected="allSpending"
          />,
          "style": Object {
            "flex": 6,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "selected": "allSpending",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "borderBottomColor": "#ccc",
                "borderBottomWidth": 1,
                "marginVertical": 20,
                "paddingBottom": 20,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
              Spent so far
            </Text>
          </Component>,
          <Styled(Component)
            style={
              Object {
                "alignContent": "center",
                "alignSelf": "center",
                "flexDirection": "row",
                "marginVertical": 20,
              }
            }
          >
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#ccc",
                    "borderBottomWidth": 4,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 30,
                  },
                  Object {
                    "backgroundColor": "#529FBE",
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ]
                }
              >
                All
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#ccc",
                    "borderBottomWidth": 4,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 30,
                  },
                  "",
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    "",
                  ]
                }
              >
                Regular
              </Text>
            </TouchableOpacity>
            <TouchableOpacity
              activeOpacity={0.2}
              onPress={[Function]}
              style={
                Array [
                  Object {
                    "backgroundColor": "#ccc",
                    "borderBottomWidth": 4,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 30,
                  },
                  "",
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    "",
                  ]
                }
              >
                Day to day
              </Text>
            </TouchableOpacity>
          </Styled(Component)>,
          <Component
            style={
              Object {
                "flex": 6,
              }
            }
          >
            <Connect(SpentDetails)
              selected="allSpending"
            />
          </Component>,
        ],
        "style": Object {
          "flex": 1,
          "justifyContent": "flex-start",
          "paddingHorizontal": 25,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                }
              }
            >
              Spent so far
            </Text>,
            "style": Object {
              "borderBottomColor": "#ccc",
              "borderBottomWidth": 1,
              "marginVertical": 20,
              "paddingBottom": 20,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Spent so far",
              "ellipsizeMode": "tail",
              "style": Object {
                "fontSize": 18,
                "fontWeight": "bold",
              },
            },
            "ref": null,
            "rendered": "Spent so far",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ccc",
                      "borderBottomWidth": 4,
                      "borderWidth": 1,
                      "height": 60,
                      "justifyContent": "center",
                      "paddingHorizontal": 30,
                    },
                    Object {
                      "backgroundColor": "#529FBE",
                    },
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 20,
                      },
                      Object {
                        "color": "#fff",
                      },
                    ]
                  }
                >
                  All
                </Text>
              </TouchableOpacity>,
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ccc",
                      "borderBottomWidth": 4,
                      "borderWidth": 1,
                      "height": 60,
                      "justifyContent": "center",
                      "paddingHorizontal": 30,
                    },
                    "",
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 20,
                      },
                      "",
                    ]
                  }
                >
                  Regular
                </Text>
              </TouchableOpacity>,
              <TouchableOpacity
                activeOpacity={0.2}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#ccc",
                      "borderBottomWidth": 4,
                      "borderWidth": 1,
                      "height": 60,
                      "justifyContent": "center",
                      "paddingHorizontal": 30,
                    },
                    "",
                  ]
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 20,
                      },
                      "",
                    ]
                  }
                >
                  Day to day
                </Text>
              </TouchableOpacity>,
            ],
            "style": Object {
              "alignContent": "center",
              "alignSelf": "center",
              "flexDirection": "row",
              "marginVertical": 20,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "All",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 20,
                      },
                      Object {
                        "color": "#fff",
                      },
                    ]
                  }
                >
                  All
                </Text>,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "backgroundColor": "#ccc",
                    "borderBottomWidth": 4,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 30,
                  },
                  Object {
                    "backgroundColor": "#529FBE",
                  },
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "All",
                  "ellipsizeMode": "tail",
                  "style": Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    Object {
                      "color": "#fff",
                    },
                  ],
                },
                "ref": null,
                "rendered": "All",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Regular",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 20,
                      },
                      "",
                    ]
                  }
                >
                  Regular
                </Text>,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "backgroundColor": "#ccc",
                    "borderBottomWidth": 4,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 30,
                  },
                  "",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Regular",
                  "ellipsizeMode": "tail",
                  "style": Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    "",
                  ],
                },
                "ref": null,
                "rendered": "Regular",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Day to day",
              "nodeType": "class",
              "props": Object {
                "activeOpacity": 0.2,
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontSize": 20,
                      },
                      "",
                    ]
                  }
                >
                  Day to day
                </Text>,
                "onPress": [Function],
                "style": Array [
                  Object {
                    "backgroundColor": "#ccc",
                    "borderBottomWidth": 4,
                    "borderWidth": 1,
                    "height": 60,
                    "justifyContent": "center",
                    "paddingHorizontal": 30,
                  },
                  "",
                ],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "Day to day",
                  "ellipsizeMode": "tail",
                  "style": Array [
                    Object {
                      "color": "#000",
                      "fontSize": 20,
                    },
                    "",
                  ],
                },
                "ref": null,
                "rendered": "Day to day",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(SpentDetails)
              selected="allSpending"
            />,
            "style": Object {
              "flex": 6,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "selected": "allSpending",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
